var/spack/repos/builtin/packages/abacus/package.py

```diff
@@ -101,3 +208,60 @@ def edit(self, spec, prefix):
 
     def install(self, spec, prefix):
         install_tree("bin", prefix.bin)
+
+
+class CMakeBuilder(cmake.CMakeBuilder):
+    def cmake_args(self):
+        spec = self.spec
+        args = [
+            self.define_from_variant("ENABLE_MPI", "mpi"),
+            self.define_from_variant("USE_OPENMP", "openmp"),
+            self.define_from_variant("USE_ELPA", "elpa"),
+            self.define_from_variant("USE_ABACUS_LIBM", "mathlib"),
```

Comment:
```py
==> abacus: Executing phase: 'cmake'
==> Error: KeyError: '"mathlib" is not a variant of "abacus"'
```
Either remove setting `USE_ABACUS_LIBM` from mathlib, set it to a fixed value or define the variant for it.

It sounds like to set if an abacus-internal libm shall be used.

Option 1, to set a fixed value, would possibly be like:
```suggestion
    self.define("USE_ABACUS_LIBM", False),
```
I guess if the system does have a "good" (performant?) libm / libc, then you may not need to use an internal libm.

But it's your expert decision what to do with it. You can also leave it to auto-detect.
If that is just the standard `libm` installed on the systems,
and you do not need to turn it off or on, your may just remove it and let abacus  decide.

BTW, are all these new variants supported by all abacus versions?

Reply:
@caic99:
- I think it should be common sense to build what you push and request review for.
- In the interest of saving face on this PR, please check that your pushes do build and work.

As that care has not yet been taken by you, I'm setting the "draft" status, to prevent spending GitLab CI compute resources on this PR until you fix the build and all requested changes are resolved.

When the PR is fixed, describe which tests you have done (including on which platforms you built it) before request reviews again, thanks a lot!